src/kernel.hpp

```diff
@@ -42,6 +42,12 @@ struct cvk_kernel : public _cl_kernel, api_object<object_magic::kernel> {
         return m_argument_values;
     }
 
+    kernel_image_channel_metadata_map& get_image_channel_metadata() {
```

Comment:
This should probably return a const reference and be a const function.

Reply:
done